uber/models.py

```diff
@@ -494,93 +494,138 @@ def lookup_attendee(self, full_name, email, zip_code):
                     return attendee[0]
             raise ValueError('attendee not found')
 
-        def next_badge_num(self, badge_type, old_badge_num):
-            # assert_badge_locked()
-            badge_type = int(badge_type)
+        def needs_badge_num(self, attendee=None, badge_type=None):
+            """
+            Takes either an Attendee object, a badge_type, or both and returns whether or not the attendee should be
+            assigned a badge number. If neither parameter is given, always returns False.
 
-            if badge_type not in c.PREASSIGNED_BADGE_TYPES:
-                return 0
+            :param attendee: Passing an existing attendee allows us to check for a new badge num whenever the attendee
+            is updated, particularly for when they are checked in.
+            :param badge_type: Allows checking for a new badge number before adding/updating the Attendee() object.
```

Comment:
I'd add a note about how if ``badge_type`` is passed then it MUST be an integer rather than a string, since this is important due to the fact that form submissions pass all parameters as strings.